File: questao15/go/conta_bancaria.go
package main

import (
    "errors"
    "fmt"
)

// Struct ContaBancaria
type ContaBancaria struct {
    titular string
    saldo   float64
}

// Método para saque com tratamento de erro
func (c *ContaBancaria) Sacar(valor float64) error {
    if valor > c.saldo {
        return errors.New(fmt.Sprintf("Saldo insuficiente: Tentativa de saque de %.2f, mas saldo é %.2f", valor, c.saldo))
    }
    c.saldo -= valor
    return nil
}

func (c *ContaBancaria) ExibirSaldo() float64 {
    return c.saldo
}

func main() {
    conta := ContaBancaria{titular: "João", saldo: 500}

    fmt.Printf("Saldo atual: %.2f\n", conta.ExibirSaldo())

    // Tentativa de saque maior que o saldo
    err := conta.Sacar(600)
    if err != nil {
        fmt.Println(err)
    }
}

File: questao15/java/ContaBancaria.java
package questao15.java;

// Classe ContaBancaria
public class ContaBancaria {
    private String titular;
    private double saldo;

    // Construtor
    public ContaBancaria(String titular, double saldo) {
        this.titular = titular;
        this.saldo = saldo;
    }

    // Método para saque
    public void sacar(double valor) throws SaldoInsuficienteException {
        if (valor > saldo) {
            throw new SaldoInsuficienteException("Saldo insuficiente: Tentativa de saque de " + valor + ", mas saldo é " + saldo);
        }
        saldo -= valor;
    }

    public double exibirSaldo() {
        return saldo;
    }

    public static void main(String[] args) {
        ContaBancaria conta = new ContaBancaria("João", 500);

        System.out.println("Saldo atual: " + conta.exibirSaldo());

        try {
            // Tentativa de saque maior que o saldo
            conta.sacar(600);
        } catch (SaldoInsuficienteException e) {
            System.out.println(e.getMessage());
        }
    }
}
